### ISSUE TITLE
Feature Request: Recipe grouping

### ISSUE THREAD

**Issue by nefrob** [casey/just#1776]
Consider allowing recipes to be aded to groups. For example, using the existing recipe attributes format:

```
[group:lint]
format:
 ...

[group:lint]
check:
  ...

[group:build]
build:
  ...
```

and listed by group when calling `just --list`, ex.

```
Available recipes:
    [lint]
    format
    check
    
    [build]
    build

    [default]
    ...
```

List groups via `just --groups`, ex.

```
Groups:
  lint
  build
  default (undecorated recipes could fall in this group or this can be dropped)
```

There might be desire to allow recipes to be part of multiple groups, so this could be considered in the implementation but should not be an initial requirement.

This might also be a way to solve: https://github.com/casey/just/issues/1652. All recipes from a group could be executed at once rather than needing to match on recipe name.

**Comment by neunenak** [comment]
This is a feature I'd like to have myself and I've started a PR working on it https://github.com/casey/just/pull/1842

**Comment by kranurag7** [comment]
Hey @neunenak 
As a user, I would like to have all import in one group. 
```bash
$ tree .
.
├── justfile
└── justfiles
    ├── docker.justfile
    ├── gh.justfile
    ├── git.justfile
```
So, as per the format above, I'll have to add attributes in each of the recipe inside `docker.justfile` to have it included inside `docker` group. But it would be easier we can group based on imports as well. 

I think this can be a separate issue since this issue is about having attribute set on an individual recipe. 

Thanks for working on it.

**Comment by casey** [comment]
I think it'll be easy to add make `[group: foo]` work on imports in a follow-up PR.

**Comment by laniakea64** [comment]
Will these recipe groups have additional functionality beyond cosmetically improving list output?

For example, if https://github.com/casey/just/issues/626 is implemented, will recipe groups be used to define which recipes could be run in parallel and which recipes have to stay sequential?

**Comment by marcbria** [comment]
It's cosmetic (helping a lot with the usability when you have long list) but not only.
[#1614](https://github.com/casey/just/issues/1614) is kind of related.

**Comment by neunenak** [comment]
https://github.com/casey/just/pull/1842 is merged now. I think the suggestion in https://github.com/casey/just/issues/1776#issuecomment-1890386390  of having there be a way to tag imports with a group would be useful and I want to try to get that in as a follow-up PR, so I think it's worth keeping this issue open.

**Comment by casey** [comment]
@neunenak I wanted to make a more specific issue, so I opened #2087 and #2086.

**Comment by epicserve** [comment]
@casey,

I didn't realize that @neunenak's PR #1842 wouldn't work with imports. #2087 is a must for our company. We have a base import that we use on all projects.
